client/logic/game/maps/scenario.gd

```diff
@@ -0,0 +1,18 @@
+class_name Scenario
+extends RefCounted
```

Comment:
Interesting that you extend the Scenario from a RefCounted. (I don't expect is a problem)

Reply:
I thought not extending RefCounted would make them extend Object, where we would have to free it. But without declaring the extending line it extends automatically of RefCounted, so it can be removed.

Reply:
```suggestion
```